torch/_decomp/decompositions.py

```diff
@@ -2010,37 +2015,53 @@ def upsample_bilinear2d(
     out_w = sym_float(output_size[1])
 
     # Calculate horizontal and vertical scaling factor
-    # TODO: Figure out if scales_h/scales_w matters here
+    # See compute_scales_value and area_pixel_compute_scale in core.
     if out_h > 1:
         if align_corners:
             h_scale_factor = (in_h - 1) / (sym_int(out_h) - 1)
         else:
-            h_scale_factor = in_h / out_h
+            if scales_h is not None and scales_h > 0:
+                h_scale_factor = 1.0 / scales_h
+            else:
+                h_scale_factor = in_h / out_h
     else:
         h_scale_factor = 0.0
 
     if out_w > 1:
         if align_corners:
             w_scale_factor = (in_w - 1) / (sym_int(out_w) - 1)
         else:
-            w_scale_factor = in_w / out_w
+            if scales_w is not None and scales_w > 0:
+                w_scale_factor = 1.0 / scales_w
+            else:
+                w_scale_factor = in_w / out_w
     else:
         w_scale_factor = 0.0
 
-    i = torch.arange(sym_int(out_h), dtype=input.dtype, device=input.device)
-    j = torch.arange(sym_int(out_w), dtype=input.dtype, device=input.device)
+    # TODO: Uses guard_int to materialize a SymPy value because the Scalar
+    # argument of arange doesn't support that.
+    i = torch.arange(guard_int(sym_int(out_h)), dtype=input.dtype, device=input.device)
+    j = torch.arange(guard_int(sym_int(out_w)), dtype=input.dtype, device=input.device)
+
+    # TODO: Not materializing these results in numerical differences.
+    h_scale_factor = guard_float(h_scale_factor)
+    w_scale_factor = guard_float(w_scale_factor)
```

Comment:
Without these `guard_float`s, cuda tests fail, but cpu is fine. I haven't looked into why. Could be a codegen issue, a triton issue, or maybe due to different dtypes being used on cuda